A potential goroutine leak in pkg/kubelet/pluginmanager/operationexecutor/operation_executor_test.go

### What happened?

If the `time.After` case is selected before the `case <-ch:`
https://github.com/kubernetes/kubernetes/blob/b6e7ee359d0e13f81cd8594820e6ec71bec3c78a/pkg/kubelet/pluginmanager/operationexecutor/operation_executor_test.go#L164-L175
which leads to the ch <- nil below get blocked, results in a goroutine leak.
https://github.com/kubernetes/kubernetes/blob/b6e7ee359d0e13f81cd8594820e6ec71bec3c78a/pkg/kubelet/pluginmanager/operationexecutor/operation_executor_test.go#L188


### What did you expect to happen?

If the `time.After` case is selected before the `case <-ch:`, the ch <- nil doesn't get blocked either.

### How can we reproduce it (as minimally and precisely as possible)?

1.Set time.After(5 * time.Second) to time.After(time.Millisecond * 100)
2.Add time.Sleep(time.Millisecond * 200) before the `ch <- nil`
https://github.com/kubernetes/kubernetes/blob/b6e7ee359d0e13f81cd8594820e6ec71bec3c78a/pkg/kubelet/pluginmanager/operationexecutor/operation_executor_test.go#L188
The two steps make the time.After happen before the `case <-ch:`.
3.Using goleak to detect the goroutine leak in the test function
https://github.com/kubernetes/kubernetes/blob/b6e7ee359d0e13f81cd8594820e6ec71bec3c78a/pkg/kubelet/pluginmanager/operationexecutor/operation_executor_test.go#L50

```
func TestOperationExecutor_UnregisterPlugin_ConcurrentUnregisterPlugin(t *testing.T) {
	defer goleak.VerifyNone(t)
```

The error report is as follows,
```
found unexpected goroutines:
        [Goroutine 7 in state chan send, with k8s.io/kubernetes/pkg/kubelet/pluginmanager/operationexecutor.startOperationAndBlock on top of the stack:
        goroutine 7 [chan send]:
        k8s.io/kubernetes/pkg/kubelet/pluginmanager/operationexecutor.startOperationAndBlock(0x517d34?, 0x0?)
        	/home/song2048/桌面/goProject/src/GoPie/testdata/realProjects/k8s.io/kubernetes/kubernetes/pkg/kubelet/pluginmanager/operationexecutor/operation_executor_test.go:183 +0x4c
        k8s.io/kubernetes/pkg/kubelet/pluginmanager/operationexecutor.(*fakeOperationGenerator).GenerateUnregisterPluginFunc.func1()
        	/home/song2048/桌面/goProject/src/GoPie/testdata/realProjects/k8s.io/kubernetes/kubernetes/pkg/kubelet/pluginmanager/operationexecutor/operation_executor_test.go:124 +0x25
        k8s.io/kubernetes/pkg/util/goroutinemap.(*goRoutineMap).Run.func1()
        	/home/song2048/桌面/goProject/src/GoPie/testdata/realProjects/k8s.io/kubernetes/kubernetes/pkg/util/goroutinemap/goroutinemap.go:119 +0xc4
        created by k8s.io/kubernetes/pkg/util/goroutinemap.(*goRoutineMap).Run
        	/home/song2048/桌面/goProject/src/GoPie/testdata/realProjects/k8s.io/kubernetes/kubernetes/pkg/util/goroutinemap/goroutinemap.go:112 +0x4d8
        
         Goroutine 8 in state chan send, with k8s.io/kubernetes/pkg/kubelet/pluginmanager/operationexecutor.startOperationAndBlock on top of the stack:
        goroutine 8 [chan send]:
        k8s.io/kubernetes/pkg/kubelet/pluginmanager/operationexecutor.startOperationAndBlock(0x0?, 0x0?)
        	/home/song2048/桌面/goProject/src/GoPie/testdata/realProjects/k8s.io/kubernetes/kubernetes/pkg/kubelet/pluginmanager/operationexecutor/operation_executor_test.go:183 +0x4c
        k8s.io/kubernetes/pkg/kubelet/pluginmanager/operationexecutor.(*fakeOperationGenerator).GenerateUnregisterPluginFunc.func1()
        	/home/song2048/桌面/goProject/src/GoPie/testdata/realProjects/k8s.io/kubernetes/kubernetes/pkg/kubelet/pluginmanager/operationexecutor/operation_executor_test.go:124 +0x25
        k8s.io/kubernetes/pkg/util/goroutinemap.(*goRoutineMap).Run.func1()
        	/home/song2048/桌面/goProject/src/GoPie/testdata/realProjects/k8s.io/kubernetes/kubernetes/pkg/util/goroutinemap/goroutinemap.go:119 +0xc4
        created by k8s.io/kubernetes/pkg/util/goroutinemap.(*goRoutineMap).Run
        	/home/song2048/桌面/goProject/src/GoPie/testdata/realProjects/k8s.io/kubernetes/kubernetes/pkg/util/goroutinemap/goroutinemap.go:112 +0x4d8
        ]
--- FAIL: TestOperationExecutor_UnregisterPlugin_ConcurrentUnregisterPlugin
```



### Anything else we need to know?

How to fix the bug:
https://github.com/kubernetes/kubernetes/blob/b6e7ee359d0e13f81cd8594820e6ec71bec3c78a/pkg/kubelet/pluginmanager/operationexecutor/operation_executor_test.go#L181
Change the unbufferd channel to a buffered channel. So that the `ch <- nil` can't get blocked in any case.
`make(chan interface{}, 1)`

### Kubernetes version

<details>

```console
$ kubectl version
# paste output here
```

</details>


### Cloud provider

<details>

</details>


### OS version

<details>

```console
# On Linux:
$ cat /etc/os-release
# paste output here
$ uname -a
# paste output here

# On Windows:
C:\> wmic os get Caption, Version, BuildNumber, OSArchitecture
# paste output here
```

</details>


### Install tools

<details>

</details>


### Container runtime (CRI) and version (if applicable)

<details>

</details>


### Related plugins (CNI, CSI, ...) and versions (if applicable)

<details>

</details>

--- a/pkg/kubelet/pluginmanager/operationexecutor/operation_executor_test.go
+++ b/pkg/kubelet/pluginmanager/operationexecutor/operation_executor_test.go
@@ -185,6 +185,9 @@ func setup(t *testing.T) (context.Context, chan interface{}, chan interface{}, O
 // This function starts by writing to ch and blocks on the quit channel
 // until it is closed by the currently running test
 func startOperationAndBlock(ch chan<- interface{}, quit <-chan interface{}) {
-	ch <- nil
-	<-quit
+	select {
+	case ch <- nil:
+		<-quit
+	case <-quit:
+	}
 }

